### PR TITLE
Add concurrency patterns:generators

### PR DESCRIPTION
**Pull request by legendtkl**
Add generator patterns. Implement with two way: closure and channel.

### CODE REVIEW

File: concurrency/generators.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
+package main
+
+import (
+	"fmt"
+)
+
+//implement generator by closure
+func FibnacciClosure() func() (ret int) {
```
Comment (tmrts) [inline concurrency/generators.go]: is this a typo or intended `Fibnacci`?

Comment (tmrts) [inline concurrency/generators.go]: `n` should suffice instead of `ret`

File: concurrency/generators.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
+package main
+
+import (
+	"fmt"
+)
+
+//implement generator by closure
```
Comment (tmrts) [inline concurrency/generators.go]: To comply with godoc conventions it should be `// FibnacciClosure implements fibonacci number geneartion` or something along those lines

File: concurrency/generator.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+# Generator Pattern
+
+[Generator](https://en.wikipedia.org/wiki/Generator_(computer_programming)) is a special routine that can be used to control the iteratoin behavior of a loop.
+
+# Implementation and Example
+With Go language, we can Implemente generator in two ways: channel and closure. Fibnacci example can be found in [generators.go](generators.go).
```
Comment (tmrts) [inline concurrency/generator.md]: `Implemente` typo

File: concurrency/generator.md [hunk 4]
```diff
@@ -0,0 +1,6 @@
+# Generator Pattern
+
+[Generator](https://en.wikipedia.org/wiki/Generator_(computer_programming)) is a special routine that can be used to control the iteratoin behavior of a loop.
```
Comment (tmrts) [inline concurrency/generator.md]: `iteratoin` typo

**Comment by tmrts** [comment]
@legendtkl Thank you for your contribution, I've left some comments, do you mind reworking the PR?

**Comment by legendtkl** [comment]
Yeah, thanks for you comments. 

Now I am going to fix.

File: concurrency/generators.go [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+//FibonacciClosure implements fibonacci number generatoin using closure
+func FibonacciClosure() func() int {
+	a, b := 0, 1
+	return func() int {
+		a, b = b, a+b
+		return a
+	}
+}
+
+//FibonacciChan implements fibonacci number generatoin using channel
```
Comment (jlburkhead) [inline concurrency/generators.go]: generation

File: concurrency/generators.go [hunk 6]
```diff
@@ -0,0 +1,43 @@
+package main
+
+import (
+	"fmt"
+)
+
+//FibonacciClosure implements fibonacci number generatoin using closure
```
Comment (jlburkhead) [inline concurrency/generators.go]: generation

**Comment by tmrts** [comment]
Thanks for addressing my comments and the PR itself, there were some conflicts, so I went ahead and merged it manually.

Please see d05638a

**Pull request closed without merging**